src/link/client-awareness/__tests__/ClientAwarenessLink.test.ts

```diff
@@ -0,0 +1,625 @@
+        "apollographql-client-version": "2.0.0",
+      });
+    });
+  });
+
+  describe("BatchHttpLink integration", () => {
+    test("is part of `BatchHttpLink`", async () => {
+      fetchMock.postOnce(uri, [response, response]);
+      const client = new ApolloClient({
+        link: new BatchHttpLink({
+          uri,
+        }),
+        cache: new InMemoryCache(),
+        name: "test-client",
+        version: "1.0.0",
+      });
+
+      void client.query({ query });
+      void client.query({ query, context: { queryDeduplication: false } });
+      await wait(10);
+
+      const headers = fetchMock.lastCall()![1]?.headers;
+      expect(headers).toStrictEqual({
+        accept: "application/graphql-response+json,application/json;q=0.9",
+        "content-type": "application/json",
+        "apollographql-client-name": "test-client",
+        "apollographql-client-version": "1.0.0",
+      });
+    });
+
+    test("is not part of `BaseBatchHttpLink`", async () => {
+      fetchMock.postOnce(uri, response);
+      const client = new ApolloClient({
+        link: new BaseBatchHttpLink({
+          uri,
+        }),
+        cache: new InMemoryCache(),
+        name: "test-client",
+        version: "1.0.0",
+      });
+
+      void client.query({ query });
+      await wait(10);
+
+      const headers = fetchMock.lastCall()![1]?.headers;
+      expect(headers).toStrictEqual({
+        accept: "application/graphql-response+json,application/json;q=0.9",
+        "content-type": "application/json",
+      });
+    });
+
+    test("`BatchHttpLink` options have priotity over `ApolloClient` options", async () => {
+      fetchMock.postOnce(uri, response);
+      const client = new ApolloClient({
+        link: new BatchHttpLink({
+          uri,
+          name: "overridden-client",
+          version: "2.0.0",
+        }),
+        cache: new InMemoryCache(),
+        name: "test-client",
+        version: "1.0.0",
+      });
+
+      void client.query({ query });
+      await wait(10);
+
+      const headers = fetchMock.lastCall()![1]?.headers;
+      expect(headers).toStrictEqual({
+        accept: "application/graphql-response+json,application/json;q=0.9",
+        "content-type": "application/json",
+        "apollographql-client-name": "overridden-client",
+        "apollographql-client-version": "2.0.0",
+      });
+    });
+
+    test("will batch requests with equal options", async () => {
+      fetchMock.postOnce(uri, [response, response]);
+      const client = new ApolloClient({
+        link: new BatchHttpLink({
+          uri,
+        }),
+        cache: new InMemoryCache(),
+        name: "test-client",
+        version: "1.0.0",
+      });
+
+      void client.query({ query, context: { queryDeduplication: false } });
+      void client.query({ query, context: { queryDeduplication: false } });
+      await wait(10);
+
+      const headers = fetchMock.lastCall()![1]?.headers;
```

Comment:
Can we add an assertion on `calls().length` or equivalent? Since this just checks `lastCall()`, its not clear to me that we actually batched anything since there might be 2 calls here.

```suggestion
      expect(fetchMock.calls()).toHaveLength(1);
      const headers = fetchMock.lastCall()![1]?.headers;
```

Reply:
More than one network request would crash the test - see
https://github.com/apollographql/apollo-client/blob/4a41eda7b0b4edea6b0333e3216d6bfcbb603347/src/link/client-awareness/__tests__/ClientAwarenessLink.test.ts#L361
```ts
      fetchMock.postOnce(uri, [response, response]);
```
above: we only mock a single response. Similary, I've made sure in all tests to either use `postOnce` or `post` with a `repeat` option.